pkg/controllers/user/cis/utils.go

```diff
@@ -107,3 +110,68 @@ func ValidateClusterBeforeLaunchingScan(cluster *v3.Cluster) error {
 	}
 	return nil
 }
+
+func GetBenchmarkVersionToUse(overrideBenchmarkVersion string, currentK8sVersion string,
+	cisConfigLister v3.CisConfigLister, cisConfigClient v3.CisConfigInterface,
+	cisBenchmarkVersionLister v3.CisBenchmarkVersionLister, cisBenchmarkVersionClient v3.CisBenchmarkVersionInterface,
+) (string, bool, error) {
+	bv := overrideBenchmarkVersion
+	shortK8sVersion := util.GetTagMajorVersion(currentK8sVersion)
+	if bv == "" {
+		cisConfigParams, err := kontainerdrivermetadata.GetCisConfigParams(
+			shortK8sVersion,
+			cisConfigLister,
+			cisConfigClient,
+		)
+		if err != nil {
+			logrus.Debugf("cisScanHandler: benchmark version not found for k8s version: %v, using default",
+				shortK8sVersion)
+			cisConfigParams, err = kontainerdrivermetadata.GetCisConfigParams(
+				"default",
+				cisConfigLister,
+				cisConfigClient,
+			)
+			if err != nil {
+				return "", false, fmt.Errorf("cisScanHandler: error fetching default cis config: %v", err)
+			}
+		}
+		bv = cisConfigParams.BenchmarkVersion
+	}
+	benchmarkInfo, err := kontainerdrivermetadata.GetCisBenchmarkVersionInfo(
+		bv,
+		cisBenchmarkVersionLister,
+		cisBenchmarkVersionClient,
+	)
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error fetching benchmark version info %v: %v",
+			bv, err)
+	}
+	if err := isClusterVersionSufficient(shortK8sVersion, benchmarkInfo.MinKubernetesVersion); err != nil {
+		return "", false, err
+	}
+	return bv, benchmarkInfo.Managed, nil
+}
+
+func isClusterVersionSufficient(shortClusterK8SVersion, benchmarkMinK8SVersion string) error {
```

Comment:
Would it be better if this returns (bool, error). Then we can tell if the version comparison succeeded or there are errors parsing versions?

Reply:
will make the change.